src/Symfony/Component/Serializer/DependencyInjection/SerializerPass.php

```diff
@@ -67,8 +70,37 @@ public function process(ContainerBuilder $container): void
             }
         }
 
+        $denormalizers = [];
+        $trueNormalizers = [];
+        /** @var Reference|Definition $definition */
+        foreach ($normalizers as $definition) {
+            $reference = null;
+            if ($definition instanceof Reference) {
+                $reference = $definition;
+                $definition = $container->getDefinition($definition);
+            }
+
+            $class = $definition->getClass();
+            if (null === $class) {
+                continue;
+            }
+            $interfaces = class_implements($class);
+            if (isset($interfaces[NormalizerInterface::class])) {
+                $trueNormalizers[] = $reference ?? $definition;
+            }
+            if (isset($interfaces[DenormalizerInterface::class])) {
+                $denormalizers[] = $reference ?? $definition;
+            }
+        }
+
         $serializerDefinition = $container->getDefinition('serializer');
-        $serializerDefinition->replaceArgument(0, $normalizers);
         $serializerDefinition->replaceArgument(1, $encoders);
+
+        if ($container->hasDefinition('serializer.normalizer') && $container->hasDefinition('serializer.normalizer')) {
```

Comment:
```suggestion
        if ($container->hasDefinition('serializer.normalizer') && $container->hasDefinition('serializer.denormalizer')) {
```

Plus, I think it'll always have these definition (so the `else` wouldn't be triggered at all)

Reply:
Thank you. 

If you are using Serializer 7.1 and FrameworkBundle 7.0 the else will be triggered.